awx/ui/client/src/notifications/shared/type-change.service.js

```diff
@@ -48,6 +49,12 @@ function (i18n) {
                 case 'webhook':
                     obj.webhook_required = true;
                     break;
+                case 'mattermost':
+                    obj.mattermost_required = true;
+                    obj.mattermost_username_required = false;
```

Comment:
It looks like username, channel and icon are fields that you want to show when the notification type is `mattermost` but they aren't required.  You shouldn't need these flags in this case.  Since it looks like the mattermost url is required `obj.mattermost_required = true;` is fine.

Reply:
Thanks I will make that change. I wasn't too familiar with this so I tried to mirror some of the other notification types